db/1.8/modules/ROOT/pages/database-connector-udt-stored-procedure.adoc

```diff
@@ -0,0 +1,328 @@
+	]
+----
++
+
+.Set Payload configuration
+image::database-oracleudt-example-4.png[Set payload configuration]
+
+[start=7]
+. Set the *Mime Type* field to `application/java`
+. Drag a *Transform Message* component to the right of *Set Payload*. +
+This component transforms the JSON objects into an object that can be mapped to the data type `PEOPLE` that the stored procedure expects as its input parameter.
+. Double click the component in the Studio canvas, and set the name to `Transform Message - Prepare UDT`.
+. In the *Output* source code view of the component, click the *Edit Current Target* button.
+. In the *Selection dialog* box, set the *Output* field to `Variable`, the *Variable name* to `in_people_tab`, and click *OK*.
+. In the *Output* field, set the following DataWeave expression that contains the `createStruct`  and `createArray` functions to create said object:
++
+[source,DataWeave,linenums]
+----
+%dw 2.0
+output application/java
+
+fun toPhoneNumberArray(phoneNumber) = Db::createArray("dbConfig", "PHONE_NUMBER_ARRAY",[Db::createStruct("dbConfig", "PHONE_NUMBER", [phoneNumber.areaCode, phoneNumber.phoneNumber])])
+fun toPerson(person) = Db::createStruct("dbConfig", "PERSON", [person.name, person.surname, person.age, toPhoneNumberArray(person.phoneNumber)])
+---
+Db::createArray("dbConfig", "PEOPLE", payload map (item, index) -> ( toPerson(item) ) )
+----
++
+
+.Transform Message configuration
+image::database-oracleudt-example-6.png[Transform Message configuration]
+
+
+[start=13]
+. Drag a *Stored procedure* operation to the right of *Transform Message*. +
+This operation calls the stored procedure using Database Connector.
+. In the *Connector configuration* field, select `dbConfig` global configuration.
+. In the *SQL Query Text* field enter `{ call proc_insert_humans(:people, :phoneBook) }`. +
+. In the *Input Parameters* field enter `{ people: vars.in_people_tab}` +
+To map the output of our transformation, which is stored in the variable called `in_people_tab` to our input parameter.
+. In the *Output Parametrs* field, select `Edit inline` and click the plus sign (*+*) to set a custom parameter:
+
+* *Key* +
+`phoneBook`
+* *Custom type* +
+`PHONE_BOOK`
+
+.Stored procedure configuration
+image::database-oracleudt-example-5.png[Stored procedure configuration]
+
+[start=18]
+. Drag a second *Transform Message* component to the right of *Stored procedure*.
+. Double click the component in the Studio canvas, and set the name to `Transform Message - response to JSON`
+. In the *Output* source code view of the component, set the following DataWeave expression:
++
+[source,DataWeave,linenums]
+----
+%dw 2.0
+output application/json
+---
+payload
+----
++
+[start=20]
+. Save the project.
+. Test the app by using the following curl command in your terminal: `curl localhost:8081/oracle`. +
+The stored procedure should return the list of phone numbers of all the people added to the table `HUMANS`.
+
```

Comment:
Wow, this one LONG task. Would it make any sense to break it into subtasks, maybe by operation?